File: typescript/src/shapes/rectangle.ts
class Rectangle implements Shape {
    width: number;
    height: number;

    constructor(width: number, height: number) {
        this.width = width;
        this.height = height;
    }

    // override the computeArea method
    computeArea(): number {
        return this.width * this.height
    }
}

export { Rectangle }

File: java/src/main/java/edu/cmu/cs/cs214/lab02/shapes/Rectangle.java
package edu.cmu.cs.cs214.lab02.shapes;

public class Rectangle implements Shape {
    // add encapsulation
    private double height;
    private double width;
    
    public Rectangle(double height, double width){
        this.height = height;
        this.width = width;
    }

    // override the getArea method
    @Override
    public double getArea() {
        return height * width;
    }
}
